source/grid.js

```diff
@@ -153,22 +172,43 @@ this.addToGridFromWorld = function( object, worldCoord ) {
 }
 
 this.removeFromGrid = function( nodeID, gridCoord ) {
-    var objects = this.getObjectsAtCoord( gridCoord );
-    var index = objects.indexOf( nodeID );
-    if ( index !== -1 ) {
-        for ( var i = index; i < objects.length - 1; i++ ) {
-            objects[ i ] = objects[ i + 1 ];
+
+    var node = this.scene.findByID( this.scene, nodeID ); 
```

Comment:
I'd prefer not to use find() / findByID() if at all possible, since it's considered to be slow. I'm keeping the function definitions the same as before (take nodeID as input, and not a pointer to the node... other parts of the code assume this to be true). Is there any easy way around this? One possibility is to pass in a pointer to the object as an optional third argument.